### PR TITLE
Added Binary Search Tree Challenge

### PR DESCRIPTION
**Pull request by anthonykyi**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/check-if-binary-search-tree.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+id: 5cc0c1b32479e176caf3b422
+title: Check if Tree is Binary Search Tree
+challengeType: 1
+---
+
+## Description
+
+<section id='description'>
+Since you already know what a binary search tree is, this challenge will establish how it is you can tell that a tree is a binary search tree or not.
+The main distinction of a binary search tree is that the nodes are ordered in an organized fashion. Nodes have at most 2 child nodes that are placed to the right and the left based on if the node's value is greater than or equal to (right) or less than (left) the parent node.
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/check-if-binary-search-tree.md]:
```suggestion
The main distinction of a binary search tree is that the nodes are ordered in an organized fashion. Nodes have at most 2 child nodes (placed to the right and/or left) based on if the child node's value is greater than or equal to (right) or less than (left) the parent node.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my one suggestion.  Also, you will need to update [meta.json](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/_meta/data-structures/meta.json) which defines the order in which the challenge will appear.

**Comment by RandellDawson** [comment]
@anthonykyi Also, you need to make sure the file is named properly.  It should actually be named:
`curriculum/challenges/english/08-coding-interview-prep/data-structures/check-if-binary-search-tree.english.md` since this is the English version of the challenge.

**Review by RandellDawson — APPROVED** [review]
LGTM

**Comment by RandellDawson** [comment]
@raisedadead Need a core team review.

**Review by raisedadead — APPROVED** [review]

**Review by Manish-Giri — APPROVED** [review]
@anthonykyi LGTM. Thank You for your contribution.

**Pull request merged**